cmd/controller-manager/app/controller_manager.go

```diff
@@ -321,3 +327,27 @@ func StartControllers(s *options.ControllerManagerServer,
 
 	select {}
 }
+
+var _ healthz.HealthzChecker = checkAPIAvailableResources{}
+
+// checkAPIAvailableResourcesServer is a HealthzChecker that makes sure the
+// Service-Catalog APIServer is contactable.
+type checkAPIAvailableResources struct {
+	serviceCatalogClientBuilder controller.ClientBuilder
+}
+
+func (c checkAPIAvailableResources) Name() string {
+	return "checkAPIAvailableResources"
+}
+
+func (c checkAPIAvailableResources) Check(_ *http.Request) error {
+	glog.Info("available resources health checker called")
+	availableResources, err := getAvailableResources(c.serviceCatalogClientBuilder)
+	if err != nil {
+		return err
+	}
+	if !availableResources[schema.GroupVersionResource{Group: "servicecatalog.k8s.io", Version: "v1alpha1", Resource: "brokers"}] {
```

Comment:
The schema.GVR? I did copy it from above. Can we do that and use it in a lookup?